From 42744387a18188ac6608aec4bdcef133d184bd16 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?G=C3=A1bor=20Gomb=C3=A1s?= <gombasg@sztaki.hu>
Date: Sat, 29 Dec 2007 18:17:52 +0100
Subject: [PATCH] Fix peer_CN check

The address of the array always evaluates to true, but the interesting
thing is its content.
---
 imap/tls.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/imap/tls.c b/imap/tls.c
index fce5929..ef85993 100644
--- a/imap/tls.c
+++ b/imap/tls.c
@@ -976,7 +976,7 @@ int tls_start_servertls(int readfd, int writefd, int timeout,
 
 	if (authid != NULL) {
 	    /* save the peer id for our caller */
-	    *authid = peer_CN ? xstrdup(peer_CN) : NULL;
+	    *authid = peer_CN[0] ? xstrdup(peer_CN) : NULL;
 	}
 	X509_free(peer);
     }
@@ -1326,7 +1326,7 @@ int tls_start_clienttls(int readfd, int writefd,
 
 	if (authid != NULL) {
 	    /* save the peer id for our caller */
-	    *authid = peer_CN ? xstrdup(peer_CN) : NULL;
+	    *authid = peer_CN[0] ? xstrdup(peer_CN) : NULL;
 	}
 	X509_free(peer);
     }
